neqo-transport/src/recovery.rs

```diff
@@ -329,6 +505,20 @@ impl LossRecovery {
             .map(|(_k, v)| v)
             .collect::<Vec<_>>();
 
+        // OnPacketAcked
+        for pkt in acked_packets.iter().filter(|pkt| pkt.in_flight) {
+            self.cc.on_packet_acked(pkt)
+        }
+
+        if !lost_packets.is_empty() {
+            self.cc.on_packets_lost(
```

Comment:
I think that there might be a bug here.  A packet can be both lost AND acknowledged.  Right now, you filter on pkt.in_flight for acknowledged packets, which is right, I think, but I don't see where that flag is set.

Reply:
I don't see how a packet could be both, in the context of a single call here. We `remove_acked()` up above, before we call `detect_lost_packets()`. Now it is true that we could flag a packet as lost when one ack is received and then mark it as acked when another ack comes in (we just started keeping lost SentPackets around for 2*RTT not long ago) but that should be handled w/o issue.

`in_flight` is now set when the SentPacket is constructed, and is true if packet is either ack-eliciting or has padding. Afaict the only non-in-flight packets are ack-only packets, right?

Reply:
Not in the context of a single call, but over time the same packet might be lost and then later "found".  This is #267.